tests/unit/utils.py

```diff
@@ -441,6 +441,9 @@ def setUp(self):
         else:
             LoggingManager().start(level=logging.WARNING)
 
+    def _addDuration(self, result, elapsed):  # For Python >= 3.12
```

Comment:
is overriding this method not just masking a real issue i.e. that addDuration() does not exist?

Reply:
It's a new function in 3.12 that subclasses are expected to implement:
https://docs.python.org/3/library/unittest.html#unittest.TestResult.addDuration

It seems for tracking testcase execution times and for running any cleanups. I thought we don't care about either of these right now.

Reply:
I believe it would be better to have a docstring stating that instead of having "pass" as the function body for future travelers.

Reply:
ack, done

Reply:
@pponnuvel shouldn't the method be called addDuration()?

Reply:
Ok I now see my confusion, addDuration is for TestResult whereas \_addDuration is for TestCase. For what its worth the commit message is a bit misleading since it is missing the "_".

Reply:
In order to avoid any forward compatibility issues, we could also define a more generic method i.e.:

```
def _addDuration(self, *args, **kwargs):  # For Python >= 3.12
...
```